static/markdown/tap-target-maintenance.md

```diff
@@ -3,38 +3,141 @@ title: "Maintaining Taps & Targets"
+
+## Best Practices
+
+### Avoid Creating a New Variant When Possible
+
+It's not uncommon in the Singer community (and the open source ecosystem in general) to see people fork a repo in order to add a single feature or bug fix but never get it merged to the upstream repo.
+This approach may solve the short term problem but requires you to take on the burden of maintaining the repo moving forward.
+By contributing back to the upstream you benefit the community and you can free yourself from maintaining your variant moving forward.
+If everyone in our community puts their effort into maintaining the same connectors then we all benefit from less work and better quality.
+
+### Favor Generically Useful vs Custom or Hardcoded Logic
+
+Sometimes it's tempting to build a connector with only your exact needs baked in. For example only adding the streams you need or hard coding organization specific logic like account names, how credentials are sourced, etc. But with a little extra effort you can make it generically useful enough to attract other community members to use and help maintain your code.
+
+- Add as much stream coverage as you can
+- All configs should flow through the config.json input file
+- No hardcoded values and logic, or at least not by default
+
+### Document Settings Well
+
+The most challenging part of using a community maintained connector is configuring it properly.
+Always make sure to document your configurations well.
+
+SDK based connectors have a [--about command option](https://sdk.meltano.com/en/latest/implementation/cli.html#about) that prints out the documented settings and descriptions from `tap.py`. MeltanoHub scrapes these settings and descriptions for building documentation so you should make sure to keep them up to date.
+
+For non-SDK based connectors you will commonly find an example config.json in the repo.
+The Meltano community has also invested in documenting these on MeltanoHub for your convenience.
+
+### Advertise It On Meltano Hub
+
+Once your connector is functioning and ready to go, make sure you get it listed on MeltanoHub (see [docs](/add-a-tap)) so others can find it.
+More users means more contributors and maintainers for your code base!
+
+### Use the Meltano Singer SDK
+
+Using the SDK allows you to reduce code, increase reliability, comply with the Singer Spec, gain features with an SDK version upgrade, and work well within the Meltano community. The SDK is maintained by Meltano and receives regular updates. Upgrading a tap built on the SDK is much easier than one that isn't.
+
+## Common Patterns
+
+Since Singer is only a spec (see Meltano's [Singer Spec docs](http://localhost:8080/singer/docs#singer-spec)) for how data should be formatted and passed between connectors there have become many common patterns, tools, and SDKs to help speed the process up.
```

Comment:
```suggestion
Since Singer is a primarily a specification for data formatting and transfer(see Meltano's [Singer Spec docs](http://localhost:8080/singer/docs#singer-spec)) there have become many common patterns, tools, and SDKs to help build connectors and pipelines more efficiently.
```